.ci/Jenkinsfile

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env groovy
+
+// Verified with Jenkins v2.164.3
+
+// TODO:
+// 1. Calculate taskset/affinity for the scripts based on total number of jenkins executors
+
+pipeline {
+    agent {label "ml-test-node-gpu"}
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'10'))
+        timeout(time: 15, unit: 'MINUTES')
+        disableConcurrentBuilds()
+    }
+
+    environment {
+        ARTEFACT_DIR = "${WORKSPACE}/artefacts"
+        NFS_WORKSPACE = "${NFS_WORKSPACE_ROOT}/ml-nccl-rdma-sharp-plugins-pr/${BUILD_NUMBER}"
+        NCCL_SRC_DIR = "${WORKSPACE}/nccl"
+        NCCL_DIR = "${NFS_WORKSPACE}/nccl"
+        NCCL_TESTS_DIR = "${NFS_WORKSPACE}/nccl-tests"
+    }
+
+    stages {
+        stage('Preparations') {
+            steps {
+                echo 'Preparations...'
+                sh 'mkdir -p ${ARTEFACT_DIR}'
+                sh 'mkdir -p ${NFS_WORKSPACE}'
+            }
+        }
+        stage('Checkout NCCL') {
+            steps {
+                sh 'mkdir -p ${NCCL_SRC_DIR}'
+                dir("${NCCL_SRC_DIR}") {
+                    git branch: 'master',
```

Comment:
Is it OK to use an external master for CI? Maybe use tag/branch to make this CI more robust.

Reply:
NCCL CI/nightly is to be established - after that we'll consider NCCL-SHARP CI with stable/master/custom/etc NCCL versions. Right now there's NCCL gitlab fork which is changed pretty rarely so it should not impact much the stability of NCCL-SHARP CI.